### PR TITLE
Openapi 3.0 [DRAFT]

### PR DESCRIPTION
**Pull request by blakepettersson**
This is an attempt to address #603. OpenAPI 3 is quite different from Swagger 2, which leads to a couple of issues with this PR.

* This PR depends on a patched `grape-swagger-entity` (which is not in this PR)
* `bundle exec rake` *DOES NOT* currently work, since there's currently an issue with running the Swagger 2 and OpenAPI 3 tests at the same time...
* Currently the Group Array Params specs as well as the Mounted spec for OpenAPI 3 are not working (that can be addressed though)

In short, given how different OpenAPI 3 is from Swagger 2, it might be worth having a separate project for this, especially since we need to have a version of `grape-swagger-entity` that's works with OpenAPI3 (which implies that the same time that it won't work for Swagger 2).
 
It was a while since I worked on this, so there are for sure some other issues with this code, which I'll try to address once I remember what I've done.

### CODE REVIEW

**Comment by swistaczek** [comment]
Great work! 🤞for future work landing in upstream!

**Comment by LeFnord** [comment]
Hi @blakepettersson … many thanks for the great work 😄 

I've created a [oapi-3](https://github.com/ruby-grape/grape-swagger/tree/oapi-3) branch, and re-target it gainst it

**Comment by LeFnord** [comment]
ok … let us mörge it

**Comment by gkirill** [comment]
Guys, sorry, any suggestions how to enable that? By default it seems to be using 2.0

**Comment by LeFnord** [comment]
use it in your Gemfile by specifying the repo and branch → https://bundler.io/guides/git.html

**Comment by eistomin** [comment]
@LeFnord , can i help you with master tree merging(ruby-grape:oapi-3 -> master or vice versa)?

**Comment by run27017** [comment]
It stills not merged into master?

**Comment by j-monteiro** [comment]
Are there any plans to merge this into master?

**Comment by EugeneIstomin** [comment]
Hi all!
Can we update & merge this branch? It would be like a 2-3 devs collaboration in upditing, merging, testing.
